platform/windows/detect.py

```diff
@@ -383,36 +490,39 @@ def configure_mingw(env):
     if os.name != "nt":
         env["PROGSUFFIX"] = env["PROGSUFFIX"] + ".exe"  # for linux cross-compilation
 
-    mingw_prefix = ""
-
-    # TODO: This doesn't seem to be working, or maybe I just have
-    # MinGW set up incorrectly. It always gives me x86_64 builds,
-    # even though arch == "x86_32" and the file name has x86_32 in it.
     if env["arch"] == "x86_32":
         if env["use_static_cpp"]:
             env.Append(LINKFLAGS=["-static"])
             env.Append(LINKFLAGS=["-static-libgcc"])
             env.Append(LINKFLAGS=["-static-libstdc++"])
-        mingw_prefix = env["mingw_prefix_32"]
+            env["x86_libtheora_opt_gcc"] = True
     else:
+        if env["arch"] == "x86_64":
+            env["x86_libtheora_opt_gcc"] = True
         if env["use_static_cpp"]:
             env.Append(LINKFLAGS=["-static"])
```

Comment:
Might be a bit clearer to do it separately IMO:
```python
if env["use_static_cpp"]:
    env.Append(LINKFLAGS=["-static"])
    if env["arch"] == "x86_32":
        env.Append(LINKFLAGS=["-static-libgcc"])
        env.Append(LINKFLAGS=["-static-libstdc++"])

if env["arch"] in ["x86_32", "x86_64"]:
     env["x86_libtheora_opt_gcc"] = True
```
Though the latter could be refactored to be done in `modules/theora/SCsub`.